### ISSUE TITLE
Multiple `Error compiling Cython file` with 3.8.0

### ISSUE THREAD

**Issue by fcannini** [msmbuilder/msmbuilder#1124]
Hello there.

I'm trying to compile msmbuilder 3.8.0 in the following environment:

OS = centos 7.6 x86_64
CC = gcc 7.3.0, 8.3.0
CYTHON = 0.29.12
NUMPY = 1.16.2
MDTRAJ = 1.9.3

When it comes to cythonize **msmbuilder/msm/_ratematrix.pyx**, the following error appears:

```
cythoning msmbuilder/msm/_ratematrix.pyx to msmbuilder/msm/_ratematrix.cpp
/opt/spack/opt/spack/linux-centos7-x86_64/gcc-7.3.0/py-cython-0.29.12-guii4wbao3lr2mxxabrosmcndougbsah/lib/python2.7/site-packages/Cython/Compiler/Main.py:369: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /tmp/root/spack-stage/spack-stage-KCroJ2/msmbuilder-3.8.0/msmbuilder/msm/_ratematrix.pyx
tree = Parsing.p_module(s, pxd, full_module_name)
Error compiling Cython file:
------------------------------------------------------------
...
if (A.shape[0] != B.shape[0]) or (A.shape[1] != B.shape[1]):
return -1
for i in range(A.shape[0]):
for j in range(A.shape[1]):
A[i, j] = A[i, j] * B[i, j]
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:113:13: Assignment to const dereference
Error compiling Cython file:
------------------------------------------------------------
...
cdef double rowsum
# T = np.dot(np.dot(V, np.diag(expwt)), U.T)
for i in range(n):
for j in range(n):
temp[i, j] = V[i, j] * expwt[j]
cdgemm_NT(temp, U, T)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:132:20: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.
Error compiling Cython file:
------------------------------------------------------------
...
On exit, out[i] contains the derivative of the `i`th eigenvalue
of K with respect to \theta_u.
"""
cdef npy_intp i
for i in range(n):
cdgemv_N(dKu, V[:, i], temp)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:162:17: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.

Error compiling Cython file:
------------------------------------------------------------
...
On exit, out[i] contains the derivative of the `i`th eigenvalue
of K with respect to \theta_u.
"""
cdef npy_intp i
for i in range(n):
cdgemv_N(dKu, V[:, i], temp)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:162:23: Memoryview 'const double[:]' not conformable to memoryview 'double[:]'.
Error compiling Cython file:
------------------------------------------------------------
...
of K with respect to \theta_u.
"""
cdef npy_intp i
for i in range(n):
cdgemv_N(dKu, V[:, i], temp)
cddot(temp, U[:, i], &out[i])
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:163:21: Memoryview 'const double[:]' not conformable to memoryview 'double[:]'.
Error compiling Cython file:
------------------------------------------------------------
...
dLdK[i, j] = counts[i, j] / T[i, j]
# out = U \left(V^T dLdK U \circ X(\lambda, t))\right) V^T
# temp2 = V^T dLdK U
cdgemm_TN(V, dLdK, temp1)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:210:18: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.
Error compiling Cython file:
------------------------------------------------------------
...
# out = U \left(V^T dLdK U \circ X(\lambda, t))\right) V^T
# temp2 = V^T dLdK U
cdgemm_TN(V, dLdK, temp1)
cdgemm_NN(temp1, U, temp2)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:211:25: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.

Error compiling Cython file:
------------------------------------------------------------
...
# temp2 =  (V^T dLdK U \circ X(w, t))
hadamard_X(w, expwt, t, n, temp2)
# dT = U \left(V^TCU \circ X(\lambda, t))\right) V^T
cdgemm_NN(U, temp2, temp1)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:217:18: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.
Error compiling Cython file:
------------------------------------------------------------
...
# temp2 =  (V^T dLdK U \circ X(w, t))
hadamard_X(w, expwt, t, n, temp2)
# dT = U \left(V^TCU \circ X(\lambda, t))\right) V^T
cdgemm_NN(U, temp2, temp1)
cdgemm_NT(temp1, V, dT)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_support.pyx:218:25: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.
Error compiling Cython file:
------------------------------------------------------------
...
lognorm = log(norm)
for i in range(n):
logpi[i] = theta[i] - lognorm
cddot(alpha, logpi, &logp)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix_priors.pyx:53:10: Memoryview 'const double[::1]' not conformable to memoryview 'double[:]'.
Error compiling Cython file:
------------------------------------------------------------
...
# dK_dtheta_u(theta, n, i, j, out=dKij)
# cdgemv_N(covar_theta, dKij, temp)
memset(&dKij[0], 0, size*sizeof(double))
memset(&temp[0], 0, size*sizeof(double))
dK_dtheta_u(theta, n, i, j, out=dKij, A=covar_theta, out2=temp)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix.pyx:539:52: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, :]'.

Error compiling Cython file:
------------------------------------------------------------
...
for u in range(size):
dK_dtheta_ij(theta, n, u, A=None, out=dKu)
dw_du(dKu, U, V, n, temp1, dlambda_dtheta[:, u])
for i in range(n):
cdgemv_N(covar_theta, dlambda_dtheta[i, :].copy(), temp2)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix.pyx:680:17: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.
Error compiling Cython file:
------------------------------------------------------------
...
dw_du(dKu, U, V, n, temp1, dlambda_dtheta[:, u])
for i in range(n):
dtau_dtheta[i, u] = dlambda_dtheta[i, u] / (w[i]**2)
for i in range(1, n):
cdgemv_N(covar_theta, dtau_dtheta[i], temp2)
^
------------------------------------------------------------
msmbuilder/msm/_ratematrix.pyx:751:17: Memoryview 'const double[:, ::1]' not conformable to memoryview 'double[:, ::1]'.
building 'msmbuilder.msm._ratematrix' extension
/opt/spack/lib/spack/env/gcc/gcc -fno-strict-aliasing -fPIC -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Imsmbuilder/src -I/opt/spack/opt/spack/linux-centos7-x86_64/gcc-7.3.0/py-numpy-1.16.2-oryyq6xlx6i3h63hwszmf6whqcjhp54s/lib/python2.7/site-packages/numpy/core/include -I/opt/spack/opt/spack/linux-centos7-x86_64/gcc-7.3.0/python-2.7.16-6m7witfusxiw625ellmavidyr7kboo3v/include/python2.7 -c msmbuilder/msm/_ratematrix.cpp -o build/temp.linux-x86_64-2.7/msmbuilder/msm/_ratematrix.o -fopenmp
cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
msmbuilder/msm/_ratematrix.cpp:1:2: error: #error Do not use this file, it is the result of a failed Cython compilation.
#error Do not use this file, it is the result of a failed Cython compilation.
^~~~~
error: command '/opt/spack/lib/spack/env/gcc/gcc' failed with exit status 1
```

TIA,
Fabricio

**Comment by mpharrigan** [comment]
Should be fixed by #1102

**Comment by fcannini** [comment]
@mpharrigan Worked like a charm, thanks!
And thank you for the quick answer.
